app/src/main/java/com/alphawallet/app/viewmodel/ImportWalletViewModel.java

```diff
@@ -40,43 +41,61 @@ public class ImportWalletViewModel extends BaseViewModel implements OnSetWatchWa
     private final KeyService keyService;
     private final AWEnsResolver ensResolver;
-    private final AnalyticsServiceType analyticsService;
 
-    private final MutableLiveData<Wallet> wallet = new MutableLiveData<>();
+    private final MutableLiveData<Pair<Wallet, ImportWalletType>> wallet = new MutableLiveData<>();
     private final MutableLiveData<Boolean> badSeed = new MutableLiveData<>();
     private final MutableLiveData<String> watchExists = new MutableLiveData<>();
-    private String importWalletType = "";
 
     @Inject
     ImportWalletViewModel(ImportWalletInteract importWalletInteract, KeyService keyService,
-                          AnalyticsServiceType analyticsService) {
+                          AnalyticsServiceType analyticsService)
+    {
         this.importWalletInteract = importWalletInteract;
         this.keyService = keyService;
         this.ensResolver = new AWEnsResolver(TokenRepository.getWeb3jService(MAINNET_ID), keyService.getContext());
-        this.analyticsService = analyticsService;
+        setAnalyticsService(analyticsService);
     }
 
-    public void onKeystore(String keystore, String password, String newPassword, KeyService.AuthenticationLevel level) {
-        importWalletType = C.AN_KEYSTORE;
+    public void onKeystore(String keystore, String password, String newPassword, KeyService.AuthenticationLevel level)
+    {
         progress.postValue(true);
 
         importWalletInteract
                 .importKeystore(keystore, password, newPassword)
                 .flatMap(wallet -> importWalletInteract.storeKeystoreWallet(wallet, level, ensResolver))
                 .subscribeOn(Schedulers.computation())
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(this::onWallet, this::onError).isDisposed();
+                .subscribe(w -> onWallet(w, ImportWalletType.KEYSTORE), this::onError).isDisposed();
     }
 
-    public void onPrivateKey(String privateKey, String newPassword, KeyService.AuthenticationLevel level) {
-        importWalletType = C.AN_PRIVATE_KEY;
+    public void onPrivateKey(String privateKey, String newPassword, KeyService.AuthenticationLevel level)
+    {
         progress.postValue(true);
         importWalletInteract
                 .importPrivateKey(privateKey, newPassword)
                 .flatMap(wallet -> importWalletInteract.storeKeystoreWallet(wallet, level, ensResolver))
                 .subscribeOn(Schedulers.computation())
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(this::onWallet, this::onError).isDisposed();
+                .subscribe(w -> onWallet(w, ImportWalletType.PRIVATE_KEY), this::onError).isDisposed();
+    }
+
+    public void onSeed(String walletAddress, KeyService.AuthenticationLevel level)
+    {
+        if (walletAddress == null)
+        {
+            progress.postValue(false);
+            Timber.e("ERROR");
```

Comment:
ERROR better be more detailed. Such as `walletAddress is null`. So that it's easier to locate in log file.

Reply:
I think this is very old code, but got highlighted when I formatted it. But awesome catch nonetheless! :)

Reply:
Done!